core/metrics/metrics.go

```diff
@@ -31,6 +32,8 @@ type Handler interface {
 	Propagator() propagation.TextMapPropagator
 	// Type returns the handler type.
 	Type() HandlerType
+	// Meter returns an otel meter provider
+	Meter() *sdkmetric.MeterProvider
```

Comment:
should return an interface so you can no-op things. You basically a need a way to have the functions people need to call from Meter() without panicing, but nothing gets passed into them